### PR TITLE
Team Building Improvements

### PR DESCRIPTION
**Pull request by MarcoPolo**
TAL at https://github.com/keybase/client/pull/14741 first (since it's commit 5715c69 in this branch)

The commits are a good description of the changes

@keybase/react-hackers

### CODE REVIEW

File: shared/util/memoize.js [hunk 1]
```diff
@@ -1,7 +1,15 @@
 // @flow
 import memoize from 'memoize-one'
 
+const memoize1Obj = (f: any) =>
+  memoize(
+    f,
+    // $FlowIssue - the type of memoize1 is wonky
+    (a, b) => Object.keys(a).length === Object.keys(b).length && !Object.keys(a).some(k => a[k] !== b[k])
```
Comment (chrisnojima) [inline shared/util/memoize.js]: why not just use shallowequal?

Comment (chrisnojima) [reply shared/util/memoize.js]: maybe prefer this to be named memoizeShallow

Comment (MarcoPolo) [reply shared/util/memoize.js]: I forgot about that lib

**Comment by MarcoPolo** [comment]
@chrisnojima Can you tal at https://github.com/keybase/client/pull/14741 too?

**Pull request merged**